apps/engmanagement/src/components/redeem-dialog.tsx

```diff
@@ -15,19 +16,28 @@ interface RedeemDialogProps {
 }
 
 const RedeemDialog = ({open = false, couponId}: RedeemDialogProps) => {
+  const {data: session} = useSession()
   const router = useRouter()
   const formik = useFormik({
     initialValues: {
-      email: '',
+      email: session?.user?.email || '',
     },
     validationSchema,
     onSubmit: async ({email}) => {
-      const purchase = await redeemFullPriceCoupon({email, couponId})
+      const {purchase, redeemingForCurrentUser} = await redeemFullPriceCoupon({
```

Comment:
checking to see if a redemption is for the logged in user because it changes how we redirect them